src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -1240,6 +1249,161 @@ def group_member_uri(group_uri, group_member, config):
+    """
+    Ingest annotation VCF into a dataset. For example, a ClinVar or gnomAD VCF.
+
+    :param dataset_uri: dataset URI
+    :param vcf_uri: VCF URI, defaults to None
+    :param search_uri: URI to search for VCF files, defaults to None
+    :param pattern: Unix shell style pattern to match when searching for VCF files,
+        defaults to None
+    :param ignore: Unix shell style pattern to ignore when searching for VCF files,
+        defaults to None
+    :param create_index: force creation of a local index file, defaults to True
+    :param config: config dictionary, defaults to None
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param register_name: name to register the dataset with on TileDB Cloud,
+        defaults to None
+    :param ingest_resources: manual override for ingest UDF resources, defaults to None
+    :param verbose: verbose logging, defaults to False
+    """
+
+    if vcf_uri is None and search_uri is None:
+        raise ValueError("vcf_uri or search_uri must be provided")
+
+    if vcf_uri and search_uri:
+        raise ValueError("vcf_uri and search_uri cannot both be provided")
+
+    logger = get_logger_wrapper(verbose)
+    logger.info("Ingesting annotation VCF into %r", dataset_uri)
+
+    if search_uri:
+        # Create and run DAG to find VCF URIs.
+        graph = dag.DAG(
+            name="vcf-find-annotations",
+            namespace=namespace,
+            mode=dag.Mode.BATCH,
+        )
+
+        vcf_uris = graph.submit(
+            find_uris_udf,
+            dataset_uri,
+            search_uri,
+            config=config,
+            include=pattern,
+            exclude=ignore,
+            verbose=verbose,
+            name="Find annotation VCF URIs ",
+            access_credentials_name=acn,
+        )
+
+        run_dag(graph)
+        vcf_uris = vcf_uris.result()
+    else:
+        vfs = tiledb.VFS()
+        if not vfs.is_file(vcf_uri):
+            raise ValueError(f"'{vcf_uri}' not found.")
+
+        vcf_uris = [vcf_uri]
+
+    # TODO: optionally filter VCFs with bcftools
+
+    # Create the DAG.
+    graph = dag.DAG(
+        name="vcf-ingest-annotations",
+        namespace=namespace,
+        mode=dag.Mode.BATCH,
+        max_workers=40,
+    )
+
+    # Add a node to create the dataset.
+    dataset_node = graph.submit(
+        create_dataset_udf,
+        dataset_uri,
+        config=config,
+        vcf_attrs=vcf_uris[0],
+        annotation_dataset=True,
+        verbose=verbose,
+        name="Create annotation dataset ",
+        access_credentials_name=acn,
+    )
+
+    # Add a node to consolidate the dataset.
+    consolidate_node = graph.submit(
+        consolidate_dataset_udf,
+        dataset_uri,
+        config=config,
+        verbose=verbose,
+        resources=CONSOLIDATE_RESOURCES,
+        name="Consolidate annotations ",
```

Comment:
Extra space at the end of name.

Reply:
The extra spaces were historically cosmetic. We can remove them now.